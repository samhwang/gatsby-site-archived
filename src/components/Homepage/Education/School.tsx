import type { EducationProps } from '../../../globals';

function Education({ institute, degree, major, duration }: EducationProps) {
  return (
    <div className="resume-item d-flex flex-column flex-md-row justify-content-between mb-5">
      <div className="resume-content">
        <h3 className="mb-0">{institute}</h3>
        <div className="subheading mb-3">{degree}</div>
        <div>{major}</div>
      </div>
      <div className="resume-date text-md-right">
        <span className="text-primary">{duration}</span>
      </div>
    </div>
  );
}

export default Education;
