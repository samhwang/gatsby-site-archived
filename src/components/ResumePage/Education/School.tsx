import type { FC } from 'react';

export interface EducationProps {
  institute: string;
  degree: string;
  major: string;
  duration: string;
}

const Education: FC<EducationProps> = ({
  institute,
  degree,
  major,
  duration,
}) => (
  <div className="resume-item d-flex flex-column flex-md-row justify-content-between mb-5">
    <div className="resume-content">
      <h3 className="mb-0">{institute}</h3>
      <div className="subheading mb-3">{degree}</div>
      <div>{major}</div>
    </div>
    <div className="resume-date text-md-right">
      <span className="text-primary">{duration}</span>
    </div>
  </div>
);

export default Education;
