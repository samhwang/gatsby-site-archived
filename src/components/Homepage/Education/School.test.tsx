/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react';
import School from './School';

describe('School Component', () => {
  it('Should render without crashing and match snapshot', () => {
    const input = {
      institute: 'RMIT University',
      degree: 'Bachelor of IT',
      major: '',
      duration: 'Oct 2016 - Oct 2017',
    };
    const tree = render(
      <School
        institute={input.institute}
        degree={input.degree}
        major={input.major}
        duration={input.duration}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
