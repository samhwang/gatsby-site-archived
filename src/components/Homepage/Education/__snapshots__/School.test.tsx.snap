// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`School Component Should render without crashing 1`] = `
<div
  className="resume-item d-flex flex-column flex-md-row justify-content-between mb-5"
>
  <div
    className="resume-content"
  >
    <h3
      className="mb-0"
    >
      RMIT University
    </h3>
    <div
      className="subheading mb-3"
    >
      Bachelor of IT
    </div>
    <div>
      
    </div>
  </div>
  <div
    className="resume-date text-md-right"
  >
    <span
      className="text-primary"
    >
      Oct 2016 - Oct 2017
    </span>
  </div>
</div>
`;
